### ISSUE TITLE
(NHL 14) Custom cellMusic Only plays once, then LCPM decode failure

### ISSUE THREAD

**Issue by camsterguy** [RPCS3/rpcs3#12309]
## Quick summary
Using a few .mp3s to make custom goal horns in NHL 14, a goal horn (or a menu music song I add) only plays once before all of the custom music fails to play again until I relaunch. 

## Details

Please describe the problem as accurately as possible.

#### 0. Make sure you're running with settings as close to default as possible
* **Do NOT enable any emulator game patches when reporting issues**
* Only change settings that are required for the game to work

#### 1. Please attach RPCS3's log.

```
E SYS: 'sys_fs_open' failed with 0x80010006 : CELL_ENOENT, “/dev_hdd0/game/BLUS31179/USRDIR/fe/ion/artassets/playerheads/p0_4000/p1309.big” [1]
U cellMusicDecode TODO: cellMusicDecodeGetDecodeStatus2(status=*0xd003a660)
U cellMusicDecode TODO: cellMusicDecodeFinalize2()
U cellSearch TODO: cellSearchFinalize()
E SYS: Stat: 'cellMusicDecodeRead2' failed with 0x8002c10b : CELL_MUSIC_DECODE_ERROR_NO_LPCM_DATA [x3093]
U PPU TODO: Unimplemented syscall syscall_988 -> CELL_OK (r3=0x1, r4=0x0, r5=0x23df3d0, r6=0x7fefffffffffffff, r7=0x23df3d0, r8=0x7f7f7f7f, r9=0x7f7f7f7f, r10=0x0)
S sys_process: Process finished
E PPU: HLE callstack savestate is not implemented!
```

#### 5. Please provide your system configuration:
* Win10 21H2
* AMD 5600x
* 1080ti
* Latest nvidia driver
* etc.

#### Please include.
* Anything else you deem to be important

**Comment by Megamouse** [comment]
This is not a log file

**Comment by camsterguy** [comment]
Sorry! will get the full one uploaded soon

**Comment by Megamouse** [comment]
Please try again and post another log if it still doesn't work

**Comment by camsterguy** [comment]
> Please try again and post another log if it still doesn't work

Hey Megamouse- I updated to the prompted version RPCS3 v0.0.23-13871-4b787b22 Alpha on launch and the same issue. First goal it plays, every goal after no dice. I quit as the second goal was scored, assuming the error is at line 39009. 

Did you want me to manually merge the change you made to cellMusicDecode.cpp as opposed to just updating the client on launch? Just let me know. thanks again so much. this is incredible to even get to play this game on PC.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9072603/RPCS3.log)

**Comment by camsterguy** [comment]
@Megamouse One thing i just noticed. After the first goal horn plays and I quit the game (but leave the rpcs3 menu up). I can't modify or delete the mp3 file i have in hdd0\music while rpcs3 is open (but only after the goal horn plays once). Maybe this issue is something related to the game reading and locking the mp3 when the goal horn is first called, and doesn't unlock it for the second + time?

**Comment by camsterguy** [comment]
> Please try again and post another log if it still doesn't work

@Megamouse Let me know if there's anything else I should try or if there's an update on this ticket. thanks!!

**Comment by Darkhost1999** [comment]
The problem is really simple to understand and should be capable of being solved.

I just got a couple questions.
When you say custom music do you mean that on an OFW PS3 you'll be able to select your own mp3's to play during these events.
Or do you mean that you've modified the game from is original content to use what you want it to use?
Does your PS3 like these modifications? Or does your PS3 have the same issues as RPCS3.

**Comment by Megamouse** [comment]
@camsterguy try with #12324
Its probably not gonna work but the log will help

**Comment by camsterguy** [comment]
> The problem is really simple to understand and should be capable of being solved.
> 
> I just got a couple questions. When you say custom music do you mean that on an OFW PS3 you'll be able to select your own mp3's to play during these events. Or do you mean that you've modified the game from is original content to use what you want it to use? Does your PS3 like these modifications? Or does your PS3 have the same issues as RPCS3.

Yes! Believe it or not, EA used to allow custom music and sounds. The OFW, real PS3 process for NHL 14 was put an mp3 on a usb, copy it to the music folder on XMB on the Ps3, then in game you can select the song or playlist and assign it to an event (like a goal song for a certain team) in the custom music settings. The game appears to treat additional folders in the hdd0/music directory as playlists, so a folder full of mp3s i called "Generic" inside the overall music folder will show in game during selection as a playlist called "generic" . It works fine on my real ps3. it plays without fail every goal, as well as the generic music in between.

This is pretty much what im trying to accomplish:
https://www.youtube.com/watch?v=2H-hvpRSHbE&t=198s

i want to use this music pack which also details the setup process
https://forums.operationsports.com/forums/ea-sports-nhl-sounds-forum/670769-cycloniacs-nhl-14-music-pack.html

Thanks!

**Comment by camsterguy** [comment]
> @camsterguy try with #12324
> Its probably not gonna work but the log will help

Does the RPCS3 auto update on launch take care of this commit or should I manually download and replace the changed files?

**Comment by camsterguy** [comment]
It's a possibility that the issue could be that the emulator grabs a mp3 file and locks it to play it and forgets to unlock it after it's done playing, because even after I close the game I cannot modify the mp3 file that plays once

**Comment by Darkhost1999** [comment]
Thanks I'm hoping custom playlists start working in my games too. I wasn't quite sure if you had modded the game or not. That's an amazing feature for your game to have. If this issue gets resolved I'll test my games again see if I can get them working with custom playlists.

**Comment by camsterguy** [comment]
> @camsterguy try with #12324 Its probably not gonna work but the log will help

Visual studio is straight up refusing to build for some reason. I'm not sure if you merged it in with the new update (I saw cellmusic stuff in the changelog) so here's the log for that. (v0.0.23-13956-086afbba Alpha)


[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9161689/RPCS3.log)

**Comment by Megamouse** [comment]
you can just download the build from the PR.
I'm not gonna tell you how to navigate there though xD

**Comment by MSuih** [comment]
Scroll to the bottom of pr page and click Show all checks -> Windows -> View details -> View more details on Cirrus -> Artifacts -> Download the larger file.

**Comment by camsterguy** [comment]
Got it! Thank you. I never knew this existed..... LOL

Loading up the game now!

**Comment by camsterguy** [comment]
> @camsterguy try with #12324 Its probably not gonna work but the log will help

line 81602 i believe (second goal), same issue. worked fine first goal. thanks everyone!
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9161778/RPCS3.log)

**Comment by Megamouse** [comment]
please try again with the latest build from https://github.com/RPCS3/rpcs3/pull/12324

**Comment by camsterguy** [comment]
> please try again with the latest build from #12324

still same issue with newest build
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9169098/RPCS3.log)

**Comment by camsterguy** [comment]
> please try again with the latest build from #12324

i just tried again with a different mp3 file, compressed, same issue- plays only once before all cellmusic interaction dies

**Comment by Megamouse** [comment]
Please try again

**Comment by camsterguy** [comment]
> Please try again

I'm using the master branch with autoupdate since it seems you merged 12324 into master- same issue (line 48940)

Is there an updated commit you wanted me to try?

Log here:
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/9202630/RPCS3.log)

Thanks again @Megamouse !!!

**Comment by clougheed14** [comment]
I noticed that when the horn (or any custom music) plays it creates a file called "music_selection_context_0.yml" in cache\playlists. I also noticed that the following pops up in the log when attempting to play a custom audio file the file the second time:

! cellMusicSelectionContext: No more tracks to play in playlist...
! Media: audio_decoder: finished playlist
U cellMusicDecode TODO: cellMusicDecodeGetDecodeStatus2(status=*0xd014a3c0)
! cellMusicDecode: cellMusicDecodeGetDecodeStatus2: status=0

and it's not creating a new "music_selection_context" .yml file.

Any time after that, it doesn't even seem like it is attempting to play/decode an audio file. Perhaps because the "playlist" is finished? Even though it is supposed to play a new custom audio file?

Also, it's not until you terminate the game until these pop up:

U cellMusicDecode TODO: cellMusicDecodeFinalize2()
U cellSearch TODO: cellSearchFinalize()

Then if you open the game back up, it will create a new file in that playlists cache with the number at the end incremented by 1, and repeat. 

Also, can confirm that the audio file that was loaded the first time can not be modified/deleted in the file explorer, as it is "being used" by rpcs3 until the game is terminated.

I'm no expert in this field, just observing, but maybe there is a problem with the NHL game not calling a function that finalizes the decode of an audio file? So it is always keeping that file in its memory as the one to play, but it is not playing anything because the clip is finished, and it is not searching for anything else because it assumes that that is the clip that was supposed to play?

Just speculating, but thought I'd post my quick testing incase it helps

**Comment by Kalkano27** [comment]
What jumps out to me right away, is that it seems to think there's only 1 track in the playlist, from the very beginning:

·! 0:00:49.095094 {PPU[0x1000015] Thread (UserMusicFramework::DecodeCo) [HLE:0x025340ec, LR:0x01a60524]} cellMusicDecode: set_decode_command(START): context: .magic='SUS', .content_type=1, .repeat_mode=0, .context_option=1, .first_track=0, **.tracks=1**, .hash='music_selection_context_0', .playlist:

·! 0:00:49.095477 {PPU[0x1000015] Thread (UserMusicFramework::DecodeCo) [HLE:0x025340ec, LR:0x01a60524]} Media: audio_decoder: **1 entries in playlist**. Start decoding...

Of course, not my tracks aren't playing at all (with an up-to-date build), and I'm getting:

·E 0:00:49.100573 {Music Decode Thread} Media: audio_decoder: Queuing error: -1094995529='Invalid data found when processing input'
·! 0:00:49.101133 {Music Decode Thread} Media: audio_decoder: stopping with error...
·! 0:00:49.101422 {Music Decode Thread} Media: audio_decoder: finished playlist
·! 0:00:49.101685 {Music Decode Thread} SIG: Thread time: 0.000000s (0.018810Gc); Faults: 0 [rsx:0, spu:0]; [soft:0 hard:0]; Switches:[vol:0 unvol:0]; Wait:[0.000s, spur:0]
·E 0:00:49.102709 {PPU[0x1000014] Thread (EAAudioCore Dac) [HLE:0x025340fc, LR:0x01a605cc]} SYS: 'cellMusicDecodeRead2' failed with 0x8002c107 : CELL_MUSIC_DECODE_ERROR_DECODE_FAILURE [1]
·E 0:00:49.108201 {PPU[0x1000014] Thread (EAAudioCore Dac) [HLE:0x025340fc, LR:0x01a605cc]} SYS: 'cellMusicDecodeRead2' failed with 0x8002c107 : CELL_MUSIC_DECODE_ERROR_DECODE_FAILURE [2]
·E 0:00:49.113705 {PPU[0x1000014] Thread (EAAudioCore Dac) [HLE:0x025340fc, LR:0x01a605cc]} SYS: 'cellMusicDecodeRead2' failed with 0x8002c107 : CELL_MUSIC_DECODE_ERROR_DECODE_FAILURE [3]

**Comment by Megamouse** [comment]
The issue is that the Qt decoder can't play certain files due to some idiotic metadata issues.
It's not really our fault, it's some windows thing.
You can probably easily play the song on linux.

**Comment by Kalkano27** [comment]
Alright, well the decode issue aside, all of the logs in here seem to show a playlist size of 1.

**Comment by Megamouse** [comment]
those logs are super old. so not really helpful

**Comment by Kalkano27** [comment]
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/10668874/RPCS3.log)

Here's one from yesterday.

**Comment by Kalkano27** [comment]
It seems that this whole problem is probably caused by the creation of the playlist.  No matter how many (or few) tracks I have in the music folder, the playlist is always 1 track long.  I manually added a second one to the file, and set it to read-only, but the emulator just created a new file with 1 track.

Version: 1.0
FileType: Music Playlist
ContentType: 1
ContextOption: 1
RepeatMode: 0
FirstTrack: 0
Tracks:
  - /dev_hdd0/music/All/Inevitable Struggle.mp3

I have a feeling that if the playlist creation was fixed, the files would play.

**Comment by WeirdnessUnfolds** [comment]
I still have the same problems with this.

**Comment by Darkhost1999** [comment]
The PS3 does not appear to make playlists inside a physical path.
You can create Playlists inside of VSH on RPCS3 just fine. But in my tests, I created on my real PS3 and used Webman mod for browsing paths where I just didn't see anything new created.
On RPCS3 I created a playlist called ASDFGHI and then used the Everything app to search for this unique Playlist folder name and no results turned up at all.
I believe what might need to be done is multiprocessing support anyway cause I don't think the games actually have any media player capabilities independently and they're just asking the PS3 interface to use its media player as apart of the system's firmware not the game.

**Comment by RaffClassic** [comment]
Still confirming that this problem is happening as of September 29th, 2024. I have the log here to see what the issue might be, because this problem has been happening for a long time now.

[RPCS3.log](https://github.com/user-attachments/files/17177731/RPCS3.log)

**Comment by lacarp99** [comment]
I am also having the same issue when playing NHL 14.

**Comment by nic2870** [comment]
Same here

**Comment by 5hole** [comment]
Actually what I noticed is that, if "custom music" is set to play anywhere (including the Main Menu), only one mp3 will play (it'll loop that specific song played) even if there are multiple mp3 files in a folder/playlist.

 Like OP said, once a song plays, any subsequent calls for "custom music" will not be read at all. Only "Default Music" will work as intended in any area of the game

**Comment by GerardS0406** [comment]
Yup, this occurs on NHL 14-NHL Legacy. Functions are not completed yet causing custom sounds to not work. Would love to see it work soon!

**Comment by digant73** [comment]
is that still an issue?

**Comment by Darkhost1999** [comment]
Same as the duplicated custom music issues. Yes

**Comment by trautamaki** [comment]
[RPCS3.log](https://github.com/user-attachments/files/25048577/RPCS3.log)
Here's a log of the behaviour.

Easier way to reproduce this is to set a menu music to one song. This song plays and loops fine. If the song is changed, the music does not play with `CELL_MUSIC_DECODE_ERROR_NO_LPCM_DATA`. In the log, I have added some extra logging to cellMusicDecode. The new menu music should be played at around
> ·E 0:01:33.590899 {PPU[0x1000013] Thread (EAAudioCore Dac) [HLE:0x025c4a5c, LR:0x01ae2678]} SYS: 'cellMusicDecodeRead2' failed with 0x8002c10b : CELL_MUSIC_DECODE_ERROR_NO_LPCM_DATA [1]

**Comment by sean6on5** [comment]
Hi - I am an NHL Legacy Modder and have been working on editing the game's native soundbank as a workaround for this problem, and I may have found something that can help your team solve this issue and open up the actual custom music function for this generation of EA Sports games.

NHL 14-LE and most other EA titles from this generation use EALayer3, which has multiple playback "location" options when the music is encoded. To make EALayer3 versions of songs, you can take .wav files and run them through a fork of EA Sound Exchange with EALayer3 support. When making these files, I found that unless they were encoded with playback location "stream", they would play once in the game and then lock the music player, _very_ similarly to what happens with the custom music player.

When encoding with the playback location set to stream, the custom songs and the game's standard music played normally. The options for other playback locations are: 'default', 'spu', 'maincpu', 'iopcpu', 'ds2dhw', 'ds3dhw', 'dsp', 'ram', and 'gigasample' -- so if there's a way to get the music to play from stream in these feature, then it may be the fix for all the EA games with missing music features.

I've attached some info that the sound exchange EXE provides, and my larger sound editing tool kit is [here ](https://drive.google.com/drive/folders/1xAR_F5UAFwT028p-Fv15N6fK0vuNg1DT?usp=drive_link) in case you'd like to reference my notes on the issue.

[sx.exe options.txt](https://github.com/user-attachments/files/25345748/sx.exe.options.txt)

**Comment by digant73** [comment]
please retest. maybe fixed

**Comment by Vasya07** [comment]
@sean6on5 Maybe we can change the in-game sounds to our without EA Trax Manager? Or is it impossible and there are no ideas how to do it?

**Comment by sean6on5** [comment]
> [@sean6on5](https://github.com/sean6on5) Maybe we can change the in-game sounds to our without EA Trax Manager? Or is it impossible and there are no ideas how to do it?

I’ve done it by reverse engineering the sound bank, however, this is only good for 1:1 replacements of songs from the original game, not the same bespoke sound editing that EA Trax allowed. So fingers crossed this eventually is added.

**Comment by digant73** [comment]
this issue could be related (maybe fixed) to PR #19478

**Comment by digant73** [comment]
please test PR #19478

**Comment by trautamaki** [comment]
> please test PR [#19478](https://github.com/RPCS3/rpcs3/pull/19478)

This does not fix the issue.
